### PR TITLE
feat(vpn): add resource VPN user group

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add resource VPN user group
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add resource VPN user group
2. make user_group_id field computed
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccUserGroup_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccUserGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccUserGroup_basic
=== PAUSE TestAccUserGroup_basic
=== CONT  TestAccUserGroup_basic
--- PASS: TestAccUserGroup_basic (129.42s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       129.489s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
  image

### CODE REVIEW

File: huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_user_group_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  password      = "Typhoeus122"
+  description   = "test"
+}
+	  
+resource "huaweicloud_vpn_user" "user3" {
+  vpn_server_id = "%[1]s"
+  name          = "%[2]s3"
+  password      = "Typhoeus123"
+  description   = "test"
+}
```
Comment (wuzhuanhong) [inline huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_user_group_test.go]:
```suggestion
resource "huaweicloud_vpn_user" "user" {
  count         = 3
  vpn_server_id = "%[1]s"
  name          = "%[2]s${count.index}"
  password      = "Typhoeus12${count.index}"
  description   = "test"
}
```

File: docs/resources/vpn_user_group.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used.
+  Changing this creates a new resource.
+
+* `vpn_server_id` - (Required, String, NonUpdatable) The VPN server ID.
```
Comment (ruwenqiang123) [inline docs/resources/vpn_user_group.md]:
```suggestion
* `vpn_server_id` - (Required, String, NonUpdatable) Specifies the VPN server ID.
```

Comment (ruwenqiang123) [reply docs/resources/vpn_user_group.md]: check other request parameter

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_user_group.go [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+func resourceUserGroupImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid ID format, must be <vpn_server_id>/<id>")
+	}
+
+	d.Set("vpn_server_id", parts[0])
+	d.SetId(parts[1])
+
+	return []*schema.ResourceData{d}, nil
```
Comment (luoping-12345) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_user_group.go:439]:
```suggestion
	mErr := multierror.Append(d.Set("vpn_server_id", parts[0]))
	d.SetId(parts[1])

	return []*schema.ResourceData{d}, Err.ErrorOrNil()
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_user_group.go [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The name of the user group.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The description of the user group.`,
+			},
+			"users": {
```
Comment (houpeng80) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_user_group.go:63]: add a hash method? if the name or desc are updated, the users should not be updated

**Pull request merged**